packages/rocketchat-irc/server/server.js

```diff
@@ -44,10 +44,13 @@ class IrcClient {
 		this.socket.setNoDelay;
 		this.socket.setEncoding('utf-8');
 		this.socket.setKeepAlive(true);
+		this.connect = this.connect.bind(this);
+		this.onConnect = this.onConnect.bind(this);
```

Comment:
Hmmm unless I'm missing something, this gets overwrote by the next line?

Reply:
Derp, I was missing something. :see_no_evil: so never mind!

Reply:
Yeah the bind there is something else to fire off other events. Maybe that function name should change?

Reply:
👍